controller/scan/image.go

```diff
@@ -193,8 +195,16 @@ type base struct {
 
 func (r *base) url(pathTemplate string, args ...interface{}) string {
 	pathSuffix := fmt.Sprintf(pathTemplate, args...)
-	url := fmt.Sprintf("%s%s", r.regURL, pathSuffix)
-	return url
+	// Parse the base URL
+	base, err := url.Parse(r.regURL)
+	if err != nil {
+		log.WithFields(log.Fields{"error": err}).Error("Failed to parse base URL")
+		return fmt.Sprintf("%s%s", r.regURL, pathSuffix)
+	}
+
+	// Use path.Join to concatenate the existing path with the end segment
+	base.Path = path.Join(base.Path, pathSuffix)
+	return base.String()
```

Comment:
A quick example here: https://go.dev/play/p/DuyTL-mAOxR
```suggestion
        return url.JoinPath(r.regURL, pathSuffix)
```